### PR TITLE
Grid update

### PR DESCRIPTION
**Pull request by stephenmwilkins**
This is part 1 of the big Grid update available now for testing.

So far I have:
- minor tweaks to requirements.txt (adding spectres and changing some versions)
- improved the code for rebinning grids.
- added a script for making the test_grid.
- made a new test grid with various changes.
- modified `grid.py` to understand this new test grid and be more generalised allowing it to be used with AGN etc. grids.
- modified `line.py` to assume air wavelengths.
- re-written and added several examples in `examples/grid`.

To do:

- adapt everything else to use this new grid structure.

## Issue Type

- [ ] Bug
- [ ] Document
- [x] Enhancement

## Checklist

- [x] I have added docstrings to all methods
- [x] I have added sufficient comments to all lines
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no pep8 errors
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
 scipy==1.9.3
-sympy==1.11.1
 unyt==2.9.2
 cmasher==1.6.3
-numba==0.56.4
 git+https://github.com/karllark/dust_attenuation.git@v0.5.dev
 dust_extinction==1.1
 nbsphinx==0.9.1
 matplotlib==3.7.1
+spectres==2.2.0
```
Comment (WillJRoper) [inline requirements.txt:18]: What is spectres and why are we changing h5py version?

Comment (stephenmwilkins) [reply requirements.txt]: spectres is a package for re-binning spectra and its used by `rebin_grid.py`. The older `h5py` version was breaking things for me.

**Comment by stephenmwilkins** [comment]
All of the parametric examples now work.

File: requirements.txt
Comment (stephenmwilkins) [inline requirements.txt:1]: Obviously this is temporary fix until we can figure out why the versioned version was breaking for me. We will probably need to figure out how to define versions for different python versions.

**Comment by stephenmwilkins** [comment]
Some of the cosmo notebooks don't work on my machine though that looks to be an external package problem.

**Review by christopherlovell — APPROVED** [review]

**Comment by christopherlovell** [comment]
I have added `sphinx-gallery` to the documentation which builds all examples in the top level `examples/` folder

**Pull request merged**